File: DATN-Fashion-Shop-BE/src/main/java/com/example/DATN_Fashion_Shop_BE/model/ProductsTranslation.java
package com.example.DATN_Fashion_Shop_BE.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Entity
@Table(name = "products_translations")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductsTranslation extends BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", columnDefinition = "NVARCHAR(MAX)", nullable = false)
    private String name;

    @Column(name = "description", columnDefinition = "NVARCHAR(MAX)")
    private String description;

    @Column(name = "material", columnDefinition = "NVARCHAR(MAX)")
    private String material;

    @Column(name = "care", columnDefinition = "NVARCHAR(MAX)")
    private String care;

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    @ManyToOne
    @JoinColumn(name = "language_id", nullable = false)
    private Language language;
}

File: FE/DATN-Fashion-Shop-FE/src/app/models/Product/ProductsTranslation.ts
import { BaseEntity} from '../BaseEntity';
import { Product} from './product';
import {Language, LanguageImpl} from '../Language';

export class ProductsTranslation extends BaseEntity {
  id: number;
  name: string;
  description?: string;
  material?: string;
  care?: string;
  product: Product;
  language: Language;

  constructor(data?: Partial<ProductsTranslation>) {
    super(data);
    this.id = data?.id || 0;
    this.name = data?.name || '';
    this.description = data?.description || '';
    this.material = data?.material || '';
    this.care = data?.care || '';
    this.product = data?.product ? new Product(data.product) : new Product();
    this.language = data?.language ? new LanguageImpl(data.language) : new LanguageImpl();
  }
}
